pkg/virt-launcher/virtwrap/device/hostdevice/generic/hostdev.go

```diff
@@ -82,16 +163,23 @@ func createHostDevicesMetadata(vmiHostDevices []v1.HostDevice) []hostdevice.Host
 	return hostDevicesMetaData
 }
 
-// validateCreationOfAllDevices validates that all specified generic host-devices have a matching host-device.
-// On validation failure, an error is returned.
-// The validation assumes that the assignment of a device to a specified generic host-device is correct,
-// therefore a simple quantity check is sufficient.
 func validateCreationOfAllDevices(genericHostDevices []v1.HostDevice, hostDevices []api.HostDevice) error {
-	if len(genericHostDevices) != len(hostDevices) {
```

Comment:
I would keep this function as it was and call it where it was. And I would introduce new function that you call in case there is actually some dra...
This would be aligned with my comment to be rather safe and have the logic by default behave as old code.